File: src/main/java/database/dao/ILocalDataDao.java
package database.dao;

import database.dto.ABaseDtoEntity;

import java.sql.SQLException;

/**
 * Created by user on 21.02.2017.
 */
public interface ILocalDataDao<T extends ABaseDtoEntity> {

    T get(int id) throws SQLException;
    void create(T entity) throws SQLException;
    void update(T entity) throws SQLException;

}

File: agentcore/src/main/java/agentcore/database/dao/ILocalDataDao.kt
package agentcore.database.dao

import agentcore.database.dto.ABaseDtoEntity

import java.sql.SQLException

/**
 * Created by user on 21.02.2017.
 */
interface ILocalDataDao<T : ABaseDtoEntity> {

    @Throws(SQLException::class)
    operator fun get(id: Int): T

    @Throws(SQLException::class)
    fun create(entity: T)

    @Throws(SQLException::class)
    fun update(entity: T)
}
